clar2wasm/src/words/print.rs

```diff
@@ -41,18 +83,17 @@ impl ComplexWord for Print {
             .global_get(generator.stack_pointer)
             .local_set(offset);
 
-        let ty = generator
-            .get_expr_type(value)
-            .ok_or_else(|| {
-                GeneratorError::TypeError("print value expression must be typed".to_owned())
-            })?
-            .clone();
-
         // Push the value back onto the data stack
         for val_local in &val_locals {
             builder.local_get(*val_local);
         }
 
+        generator.ensure_work_space(ignore_notype(&ty)?.max_serialized_size().map_err(|_| {
```

Comment:
Ah damn, that was so simple! I always thought that function would fail if the Value didn't fit in a Clarity buff!

However, I wonder if it would be possible to create a correct Value with a serialized size bigger than u32::MAX?

Reply:
Thinking more about it, I'm not a big fan of this method. The `print` function will never fail for any valid `Value` in the interpreter, but here we rely on a function that could make the compilation fail.

Reply:
I like 2 also.

My original idea would have been to implement some kind of "light typing system" (exactly like sip5-serialization, but without the values) to store a type as bytes. Then we can pass the original value in memory and this type to the print function. This would have been light in memory. I can understand that people wouldn't want to deal with such a big project though.

Reply:
Uhm, why cannot we do that with `type_string(&ty)`? As first step, then we could further optimize the representation (or just gzip it 😂 ).

I'm assuming the string repr of a type is << than the value, ofc I may be missing a lot of context.

Reply:
I don't think the string repr of a type would be smaller. There are a lot of structs inside structs inside structs...etc. But the problem remain: we'd have to parse the type string (or am I missing something? I need holidays).

In the meantime, I think 2 was a cool solution :)

Reply:
Yes, but that's already available with `clarity::vm::ast::parse` and `TypeSignature::parse_type_repr` (since it would happen on the rust side, in the linked fn, IIUC). Anyway, implemented the number 2 for now :)

Reply:
ah nice, I had no idea :o 
Could you open an optimization issue and describe this idea?

Reply:
https://github.com/stacks-network/clarity-wasm/issues/490